modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -372,6 +388,20 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
 
   // endregion
 
+  /**
+   * Adds a signature to the transaction.
+   *
+   * @param publicKey public key that produced the signature
+   * @param signature raw signature as a hex encoded Buffer
+   */
+  addSignature(publicKey: BasePublicKey, signature: Buffer): void {
```

Comment:
I think the signature should be: `signature(publicKey: BasePublicKey, signature: string)` and then you should validate both parameters are valid pubs and hex strings.

Also, `publicKey: BasePublicKey` makes it unnatural to pass the pub since you have to craft an object like `{ pub: ... }` instead of passing the pub. How can this be fixed?

Reply:
Honestly, we have no use for the public key in our addSignature implementation. It's needed for sol when adding a signature but not dot. 

Not sure what the best approach here is.

Reply:
I think it's fine leaving at as a buffer and we can convert it to hex when we do the actual signing. We know that if we do `signature.toString('hex')` that it will convert to a valid hex. 

We will sign with this hex and I now updated the code where we sign to throw a more descriptive message if the signing fails.

I think we need to give this addSignature method some more though. I can revisit this when I make the changes to sol/dot and to simply use `signature` instead of `addSignature`. Im thinking we can overload this method as well and have one with just a signature since we dont need the public key.

Reply:
@argjv @lcovar i don't feel too strongly about this one way or the other.

i'm open to changing `publicKey` to type `string` since it makes the code a little easier.

also open to changing `signature: Buffer` to `signatureHex: string`. each coin and their SDK will likely expect a slightly different format. the overhead of re-encoding isn't a concern. just need to be extremely clear what the input is.

Reply:
You should always be able to verify a signature if you have the pub key. We can add a `TODO` and figure it out in another PR.
Another option is to make it optional (preferred).

Regarding the buffer vs string hex, we've preferred strings in the past because they are easier to read and pass around as data. It saves you from having to deal with extra libs too (in your tests you are doing `Buffer.from(signature, 'hex')` to be able to use this method when the string hex is what you are storing because it is easier)